### ISSUE TITLE
New GXVOX asserts on exit

### ISSUE THREAD

**Issue by ianneilmacleod** [GeosoftInc/gxpy#54]
- [x] fixed and verified in 9.4
- [ ] verified in 9.3.1 RC
- [ ] update [known issues](https://geosoftgxdev.atlassian.net/wiki/spaces/GXD93/pages/136151168/9.3.0+Known+Issues)

Code:
```
import geosoft.gxpy.gx as gx
import geosoft.gxpy.vox as gxvox
import numpy as np
gx = gx.GXpy()
data = np.zeros(3 * 4 * 6).reshape((6, 4, 3))
with gxvox.Vox.new("test_new", data=data, temp=True) as vox:
    print(vox.nx, vox.ny, vox.nz) # print to show we got here
```
[abort_930_126_2017_11_29_07_43_22.log](https://github.com/GeosoftInc/gxpy/files/1513823/abort_930_126_2017_11_29_07_43_22.log)

**Comment by ianneilmacleod** [comment]
_temp=True_ would case the geosoft_voxel file to be deleted.  Perhaps this is at odds with the cache refresh?  Testing again as follows, with test_new_geosoft_voxel not existing:
```
import geosoft.gxpy.gx as gx
import geosoft.gxpy.voxset as gxvox
gx = gx.GXpy()
with gxvox.Voxset.new("test_new", dimension=(35, 50, 12), overwrite=True) as vox:
    print(vox.nx, vox.ny, vox.nz)
```
[abort_930_126_2017_11_29_08_07_47.log](https://github.com/GeosoftInc/gxpy/files/1513885/abort_930_126_2017_11_29_08_07_47.log)

**Comment by jbgeosoft** [comment]
This has been fixed in 9.4 builds and should be merged into 9.3.1 too. Keeping open for now to verify once 9.3.1 is ready.

**Comment by ianneilmacleod** [comment]
@josephcaluag please verify that this fix made it to 9.3.1.  Just check below once you have checked this.

- [ ] @josephcaluag verifued

**Comment by ianneilmacleod** [comment]
@josephcaluag 
9.3.1 build 26 (2018-1-2) by running test_vox_diosplay.py still fails.
9.4.0 build 12589 (2018-1-2) works.

Symptom is that the system stops immediately upon printing "GX close":

```script: test_vox_display.py
project path: C:\Development\github\gxpy\geosoft\gxpy\tests\results\test_vox_display.py\
user path: C:\Program Files\Geosoft\Desktop Applications 9\user\
GX open
..F...GX close
```
Switching to 9.4.1 I get the expected close, as follows:

```script: test_vox_display.py
project path: C:\Development\github\gxpy\geosoft\gxpy\tests\results\test_vox_display.py\
user path: C:\Program Files\Geosoft\Desktop Applications 9 - Testing\user\
GX open
......GX close

----------------------------------------------------------------------
Ran 6 tests in 22.178s

OK

**Comment by ianneilmacleod** [comment]
**NOTE** the sample code that asserts now works, but test_vox_display.py does not.

**Comment by ianneilmacleod** [comment]
Here is a sample program - stops prematurely in 9.3.1, but completes correctly in 9.4:

```
import os
import numpy as np
import geosoft.gxapi as gxapi
import geosoft.gxpy as gxpy
import geosoft.gxpy.vox as gxvox

os.environ['GEOSOFT_TEST_MODE'] = '1'
os.environ['GEOSOFT_TESTSYSTEM_MODE'] = '1'

# set to use 9.3.1
os.environ['GX_GEOSOFT_BIN_PATH'] = 'C:\\Program Files\\Geosoft\\Desktop Applications 9\\bin'

# set to use 9.4
# os.environ['GX_GEOSOFT_BIN_PATH'] = 'C:\\Program Files\\Geosoft\\Desktop Applications 9 - Testing\\bin'

gxc = gxpy.gx.GXpy(log=print, res_stack=6, max_warnings=8, suppress_progress=True)
data = np.zeros(3 * 4 * 6).reshape((6, 4, 3))
with gxvox.Vox.new("test_new", data=data, temp=True) as vox:
    gxvoxd = gxapi.GXVOXD.create(vox.gxvox, '', 0, gxapi.rDUMMY)
    gxvoxd = None
```

**Comment by ianneilmacleod** [comment]
replaced by #71
